src/controls/MenuBar.qml

```diff
@@ -411,9 +413,16 @@ MenuBarPrivate {
             Repeater {
                 id: itemsRepeater
                 model: root.menus
+
                 Loader {
                     id: menuItemLoader
 
+                    property alias selected: opts.selected
+                    onSelectedChanged: {
+                       if(selected && menuItemLoader)
```

Comment:
Not sure about "menuItemLoader" check. Usually this check should always return true when the slot is called.